engine/resource/wscript

```diff
@@ -50,26 +51,63 @@ def build(bld):
 def kill_proc(proc):
     proc.kill()
 
-def shutdown(self):
-    if not TestContext.is_valid(test_context) or waflib.Options.options.skip_build_tests or waflib.Options.options.skip_tests:
-        return
+def _get_local_ip(self):
+    hostname = socket.gethostname()
+    local_ip = socket.gethostbyname(hostname)
+    return local_ip
+
+def _start_server(self, local_ip, port, configname=None):
+    configfile = waf_dynamo.create_test_server_config(self, port=port, ip=local_ip, config_name=configname)
 
-    proc = subprocess.Popen(["python", "-m", "http.server", "6123"],
+    proc = subprocess.Popen(["python", "-m", "http.server", str(port)],
                             shell = False,
                             cwd = "build/src/test")
-    atexit.register(kill_proc, proc)
+    return proc, configfile
 
+def _wait_for_server(self, local_ip, port):
+    timeout = 5 # seconds
     start = time.time()
     while True:
-        if time.time() - start > 5:
-            error('HTTP server failed to start within 5 seconds')
+        if time.time() - start > timeout:
+            error('HTTP server failed to start within %d seconds' % timeout)
             sys.exit(1)
         try:
-            urllib.request.urlopen('http://localhost:6123')
+            urllib.request.urlopen('http://%s:%d' % (local_ip, port))
             break
         except urllib.error.URLError:
             print('Waiting for HTTP testserver to start...')
             sys.stdout.flush()
             time.sleep(0.5)
 
-    waf_dynamo.run_tests(test_context, valgrind = True)
+def _remove_config_file(path):
+    os.remove(path)
+
+def server(self):
```

Comment:
When debugging the unit tests, you can now start a standalone server that it can use, with `waf server`